app/bundles/EmailBundle/Model/TransportType.php

```diff
@@ -4,81 +4,241 @@
 
 class TransportType
 {
+    const TRANSPORT_ALIAS = 'transport_alias';
+
+    const FIELD_HOST     = 'field_host';
+    const FIELD_PORT     = 'field_port';
+    const FIELD_USER     = 'field_user';
+    const FIELD_PASSWORD = 'field_password';
+    const FIELD_API_KEY  = 'field_api_key';
+
+    /**
+     * @var array
+     */
+    private $transportTypes = [
+        'mautic.transport.amazon'       => 'mautic.email.config.mailer_transport.amazon',
+        'mautic.transport.elasticemail' => 'mautic.email.config.mailer_transport.elasticemail',
+        'gmail'                         => 'mautic.email.config.mailer_transport.gmail',
+        'mautic.transport.mandrill'     => 'mautic.email.config.mailer_transport.mandrill',
+        'mautic.transport.mailjet'      => 'mautic.email.config.mailer_transport.mailjet',
+        'smtp'                          => 'mautic.email.config.mailer_transport.smtp',
+        'mail'                          => 'mautic.email.config.mailer_transport.mail',
+        'mautic.transport.postmark'     => 'mautic.email.config.mailer_transport.postmark',
+        'mautic.transport.sendgrid'     => 'mautic.email.config.mailer_transport.sendgrid',
+        'mautic.transport.sendgrid_api' => 'mautic.email.config.mailer_transport.sendgrid_api',
+        'sendmail'                      => 'mautic.email.config.mailer_transport.sendmail',
+        'mautic.transport.sparkpost'    => 'mautic.email.config.mailer_transport.sparkpost',
+    ];
+
+    /**
+     * @var array
+     */
+    private $showHost = [
+        'smtp',
+    ];
+
+    /**
+     * @var array
+     */
+    private $showPort = [
+        'smtp',
+    ];
+
+    /**
+     * @var array
+     */
+    private $showUser = [
+        'mautic.transport.mailjet',
+        'mautic.transport.sendgrid',
+        'mautic.transport.elasticemail',
+        'mautic.transport.amazon',
+        'mautic.transport.postmark',
+        'gmail',
+        // smtp is left out on purpose as the auth_mode will manage displaying this field
+    ];
+
+    /**
+     * @var array
+     */
+    private $showPassword = [
+        'mautic.transport.mailjet',
+        'mautic.transport.sendgrid',
+        'mautic.transport.elasticemail',
+        'mautic.transport.amazon',
+        'mautic.transport.postmark',
+        'gmail',
+        // smtp is left out on purpose as the auth_mode will manage displaying this field
+    ];
+
+    /**
+     * @var array
+     */
+    private $showApiKey = [
+        'mautic.transport.sparkpost',
+        'mautic.transport.mandrill',
+        'mautic.transport.sendgrid_api',
+    ];
+
+    /**
+     * @param $serviceId
+     * @param $translatableAlias
+     * @param $showHost
+     * @param $showPort
+     * @param $showUser
+     * @param $showPassword
+     * @param $showApiKey
```

Comment:
strings?

Reply:
I'm not sure based on implementation. Petr F. asked same so I tagged Alan there - it's his code.